packages/provider/src/eip191exceptions.ts

```diff
@@ -0,0 +1,123 @@
+import { ethers } from 'ethers'
+
+export function messageIsExemptFromEIP191Prefix(message: Uint8Array, origin: string | undefined): boolean {
+  return EIP_191_PREFIX_EXCEPTIONS.some(e => e.predicate(message, origin))
+}
+
+const EIP_191_PREFIX_EXCEPTIONS: Array<{
+  name: string
+  predicate: (message: Uint8Array, origin?: string) => boolean
+}> = [
+  // NOTE: Decentraland does not support 191 correctly.
+  { name: 'Decentraland Exception', predicate: (_message, origin) => origin === 'https://play.decentraland.org' },
```

Comment:
Maybe if we have the advanced predicate we could also validate if the message from DCL is "safe", we could do this by validating that the message DCL is asking us to sign is actually a text message and not an encoded Sequence transaction

Reply:
sure, can easily add a requirement that the message decodes as UTF-8 - could even make it a string matcher rather than an origin matcher - does DCL require us to do messages other than the login sign 191-exempted?

Reply:
I don't think they do, and having those additional checks would limit a lot the kind of damage DCL can do.